example-app/app/Http/Controllers/CityController.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\City;
+use App\Models\Airline;
+
+class CityController extends Controller
+{
+    public function index(Request $request)
+    {
+        $query = City::withCount(['departureFlights', 'arrivalFlights'])
+            ->filter($request->only('airline_id'));
+
+        if ($request->has('sort') && $request->has('order')) {
+            $query->orderBy($request->sort, $request->order);
+        }
```

Comment:
use ->when() for apply conditionals to the query

https://laravel.com/docs/11.x/queries#conditional-clauses